x-pack/plugins/security_solution/public/management/cypress/e2e/response_actions/response_console.cy.ts

```diff
@@ -228,48 +242,163 @@ describe('Response console', { tags: ['@ess', '@serverless', '@brokenInServerles
+          false
+        ).then((data) => {
+          ruleId = data.id;
+          ruleName = data.name;
         });
+      });
 
-        cy.task('readZippedFileContentOnEndpoint', {
-          hostname: createdHost.hostname,
-          path: `${homeFilePath}/upload.zip`,
-          password: 'elastic',
-        }).then((unzippedFileContent) => {
-          expect(unzippedFileContent).to.equal(fileContent);
-        });
+      after(() => {
+        if (ruleId) {
+          cleanupRule(ruleId);
+        }
+      });
+
+      it('should open responder from alert details flyout', () => {
+        waitForEndpointListPageToBeLoaded(createdHost.hostname);
+        toggleRuleOffAndOn(ruleName);
+        visitRuleAlerts(ruleName);
+        closeAllToasts();
+        openAlertDetailsView();
+
+        openResponderFromEndpointAlertDetails();
+        cy.getByTestSubj('consolePageOverlay-layout-titleHolder').should('exist');
+      });
+
+      it('should open responder from timeline view alert details flyout', () => {
+        waitForEndpointListPageToBeLoaded(createdHost.hostname);
+        toggleRuleOffAndOn(ruleName);
+        visitRuleAlerts(ruleName);
+        closeAllToasts();
+
+        openInvestigateInTimelineView();
+        openAlertDetailsView();
+        openResponderFromEndpointAlertDetails();
+        verifyResponderIsOpen();
       });
     });
 
-    it('"execute --command" - should execute a command', () => {
-      waitForEndpointListPageToBeLoaded(createdHost.hostname);
-      openResponseConsoleFromEndpointList();
-      inputConsoleCommand(`execute --command "ls -al ${homeFilePath}"`);
-      submitCommand();
-      waitForCommandToBeExecuted('execute');
+    describe('From Cases', () => {
+      let ruleId: string;
+      let ruleName: string;
+      let caseId: string;
+
+      const caseOwner = 'securitySolution';
+
+      before(() => {
+        loadRule(
+          { query: `agent.name: ${createdHost.hostname} and agent.type: endpoint` },
+          false
+        ).then((data) => {
+          ruleId = data.id;
+          ruleName = data.name;
+        });
+        loadCase(caseOwner).then((data) => {
+          caseId = data.id;
+        });
+      });
+
+      after(() => {
+        if (ruleId) {
+          cleanupRule(ruleId);
+        }
+        if (caseId) {
+          cleanupCase(caseId);
+        }
+      });
+
+      it('should open responder', () => {
+        waitForEndpointListPageToBeLoaded(createdHost.hostname);
+        toggleRuleOffAndOn(ruleName);
+        visitRuleAlerts(ruleName);
+        closeAllToasts();
+        openAlertDetailsView();
+        addAlertToCase(caseId, caseOwner);
+
+        // visit case details page
+        cy.intercept('GET', `/api/cases/${caseId}/user_actions/_find*`).as('case');
+        loadPage(`${APP_CASES_PATH}/${caseId}`);
+        openCaseAlertDetails(caseId);
+
+        cy.wait('@case', { timeout: 30000 }).then(({ response: res }) => {
```

Comment:
it's better to first wait on the alias and next write an assertion, https://docs.cypress.io/guides/guides/network-requests#Assertions

Reply:
755f87d4fa66036dd3d05198db56122e6059c852